### PR TITLE
build: replace travis with github ci

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

**Comment by dajohi** [comment]
https://github.com/dajohi/btcutil/runs/670938539

File: goclean.sh [hunk 1]
```diff
@@ -9,40 +9,29 @@
...
+      --enable=gosimple \
+      --enable=unconvert
+  )
 done
-
-go tool cover -func profile.cov
-
-# To submit the test coverage result to coveralls.io,
-# use goveralls (https://github.com/mattn/goveralls)
-# goveralls -coverprofile=profile.cov -service=travis-ci
```
Comment (onyb) [inline goclean.sh:48]: Just to be sure, are we still performing code-coverage checks?

File: goclean.sh [hunk 2]
```diff
@@ -9,40 +9,29 @@
...
---enable=golint \
---enable=vet \
---enable=gosimple \
---enable=unconvert \
---deadline=120s ./... | grep -v 'ExampleNew' 2>&1 | tee /dev/stderr)"
-env GORACE="halt_on_error=1" go test -race ./...
+for i in $(find . -name go.mod -type f -print); do
+  module=$(dirname ${i})
+  echo "==> ${module}"
+  #env GORACE=halt_on_error=1 go test -race ${module}/...
```
Comment (onyb) [inline goclean.sh:19]: I think you forgot to run the tests here.

**Comment by onyb** [comment]
@dajohi Any update on this? If you would prefer, I can take this over for the last fixes.

**Comment by dajohi** [comment]
@onyb updated

File: goclean.sh [hunk 3]
```diff
@@ -9,40 +9,32 @@
...
+    -e 's,^/,,' -e 's,/v[0-9]+$,,')
+  if [ -z "$MODNAME" ]; then
+    MODNAME=.
   fi
-fi
-done
 
-go tool cover -func profile.cov
+  # run tests
+  (cd $MODNAME && env GORACE=halt_on_error=1 go test -race)
```
Comment (onyb) [inline goclean.sh:27]: This is missing the test coverage part. You just need to collect the coverage report for each module and add the following to the GitHub action YAML  file:

```
      - name: Send coverage
        uses: shogo82148/actions-goveralls@v1
        with:
          path-to-profile: profile.cov
```

**Review by onyb — CHANGES_REQUESTED** [review]
Last two remarks:
* Test coverage is missing.
* Maybe it makes sense  to bump the Go version in `go.mod` to **1.15** since you updated the version on GitHub Action.

**Comment by onyb** [comment]
@dajohi Recent changes look good to me, but the [build](https://github.com/dajohi/btcutil/runs/1051935634) on your fork appears to be failing.

**Comment by Roasbeef** [comment]
Needs a rebase.

**Comment by dajohi** [comment]
@onyb @Roasbeef ready for review..

File: goclean.sh [hunk 4]
```diff
@@ -9,40 +9,38 @@
...
+    MODNAME=.
   fi
-fi
-done
 
-go tool cover -func profile.cov
+  # run tests
+  (cd $MODNAME &&
+    echo "mode: atomic" > profile.cov && \
+    env GORACE=halt_on_error=1 go test -race -covermode=atomic -coverprofile=profile.tmp && \
```
Comment (onyb) [inline goclean.sh:29]: I think it is necessary to add `./...` here:
```suggestion
    env GORACE=halt_on_error=1 go test -race -covermode=atomic -coverprofile=profile.tmp ./... && \
```

You can verify it by doing:

```
$ cd btcutil
$ go test -v | wc -l
53
$  go test -v ./... | wc -l
179
```

**Review by onyb — CHANGES_REQUESTED** [review]

**Review by onyb — APPROVED** [review]
Finally! 🚀

**Review by jcvernaleo — APPROVED** [review]
OK

**Pull request merged**